code/modules/mob/living/living.dm

```diff
@@ -864,7 +864,10 @@
 		on_fire = 1
 		src.visible_message("<span class='warning'>[src] catches fire!</span>", \
 						"<span class='userdanger'>You're set on fire!</span>")
-		src.set_light(3)
+		var/lightcolor = LIGHT_COLOR_FIRE
+		if(light_power && light_range && light_color) //make sure the light is actually active!
+			lightcolor = BlendRGB(LIGHT_COLOR_FIRE, light_color, 0.6)
+		set_light(3, l_color = lightcolor)
```

Comment:
no, you can't just set the mob's light.

Reply:
What the hell would I do then?

Reply:
Make a fake object and put it in the mob.

Reply:
that's retarded but ok